### PR TITLE
`General`: Add tests for authentication related components

### PR DESCRIPTION
**Pull request by marc-fett**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

# Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

This PR significantly expands and improves the client-side test coverage for the **User Management** and **Authentication** module.

The main goal is to ensure that all authentication-related components behave predictably, handle edge cases, and interact correctly with the newly standardized shared service mocks under `util/`.
This PR also prepares the upcoming client test workflow by replacing outdated mocks, fixing missing mock methods, correcting inconsistent behavior, and ensuring all components use the unified mocks.

### Description

This PR introduces the following major changes￼:

1. Updated core component logic

- Fixed header imports and updated translateService.getCurrentLang() usage.
- Reworked openLoginDialog() to use the unified AuthDialog mock structure.
- Improved routing + professor/applicant detection logic in HeaderComponent.
- Normalized account service behavior (authorities, update functions, loaded state).

2. Added missing functionality to shared mocks
All mocks in `/util` received necessary method and signal extensions required by authentication components:

- AccountServiceMock: Added authorities, setAuthorities, updateUser, updatePassword, improved hasAnyAuthority logic.
- AuthOrchestratorServiceMock: Added registerProgress, totalRegisterSteps, open(), close(), signal wiring, error handling.
- RouterMock: Added url, events, routerState.snapshot, nested route simulation.
- TranslateMock: Added getCurrentLang() to replace deprecated currentLang.
- DialogServiceMock, DynamicDialogConfigMock, HttpClientMock, ToastServiceMock: All extended to support missing or newly used calls.

3. Added complete test suites for all auth components

New and comprehensive tests for:
- AuthDialogService
- AuthCardComponent
- HeaderComponent
- Login component
- Registration component
- ProfileComponent
- SidebarComponent
- SidebarButtonComponent

All important behaviors are now covered, including:
- Dialog open/close behavior (including idempotency & event-based auto-close)
- Login flow (email → password → OTP)
- TUM SSO redirection logic
- Registration flow (OTP, profile, password steps)
- Sidebar role-based category rendering
- Route authority handling + fallback behavior
- Translation fallback logic
- UI template rendering (logo, language buttons, inputs)
- All edge cases (empty email, failed updates, unsupported languages, etc.)

4. Improved consistency across codebase

- Reorganized import statements in components to ensure consistent ordering.
- Removed deprecated usages inside components (currentLang).
- Updated form behavior in Profile and Registration components.
- Updated all tests to rely on unified shared mocks instead of inline mocks.

#### Code Review

- [x] Code Review 1

### Test Coverage

| Class/File                             | Line Coverage | Confirmation (assert/expect) |
|----------------------------------------|--------------:|-----------------------------:|
| auth-dialog.service.ts                 | 100%          | ✅                           |
| header.component.ts                    | 100%          | ✅                           |
| login.component.ts                     | 100%          | ✅                           |
| registration.component.ts              | 100%          | ✅                           |
| profile.component.ts                   | 100%          | ✅                           |
| sidebar.component.ts                   | 100%          | ✅                           |
|sidebar-button.component.ts                 | 100%          | ✅                           |
| auth-card.component.ts                 | 100%          | ✅                           |

### CODE REVIEW

**Review by Bofan-Zhu — COMMENTED** [review]
Thanks for the tests. Could you add describe blocks to the test files where it might make sense to group the tests (Just for readability) and fix the codacy errors if applicable?

**Review by sachmii — APPROVED** [review]
thank you 👍

File: src/test/webapp/app/shared/components/organisms/login/login.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    });
+
+    it('should return false and show error on failed email login', async () => {
+      vi.spyOn(authFacade, 'loginWithEmail').mockResolvedValue(false);
+      const showErrorSpy = vi.spyOn(toastService, 'showError');
+
+      const result = await component.onEmailLogin('test@example.com', 'password');
+
+      expect(result).toBe(false);
+      expect(showErrorSpy).toHaveBeenCalled();
```
Comment (az108) [inline src/test/webapp/app/shared/components/organisms/login/login.component.spec.ts]: please avoid toHaveBeenCalled() and use something more concrete
